api.ts

```diff
@@ -372,6 +372,34 @@ export type ParameterOptions<T extends ParameterType> = Omit<ParamDef<ParameterT
     : undefined;
 };
 
+type UnionParameterOptions =
```

Comment:
I did not want to do this horrendous union, but when I tried do this with generics, TypeScript would not properly infer the return value of `makeParameter()`:

```typescript
makeParameter<T extends ParameterType, O extends ParameterOptions<T>>(options: O): DeriveStuffFromO<O> 
```

The output of this was not a parameter bound to a specific value of ParameterType as I think it should be, but rather a parameter bound to the non-specific `ParameterType` enum class, which breaks all specific inference downstream.

Reply:
it seems to work for me removing `UnionParameterOptions`. just pushed a commit for demonstration. can you take a look if that makes sense?